lib/node_modules/@stdlib/stats/mskmax/lib/main.js

```diff
@@ -0,0 +1,101 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isNdarrayLike = require( '@stdlib/assert/is-ndarray-like' );
+var format = require( '@stdlib/string/format' );
+var maybeBroadcastArrays = require( '@stdlib/ndarray/base/maybe-broadcast-arrays' );
+var base = require( './base.js' );
+
+
+// MAIN //
+
+/**
+* Computes the maximum value along one or more ndarray dimensions according to a mask.
+*
+* @param {ndarray} x - input ndarray
+* @param {ndarray} mask - mask ndarray
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform a reduction
+* @param {boolean} [options.keepdims=false] - boolean indicating whether the reduced dimensions should be included in the returned ndarray as singleton dimensions
+* @param {string} [options.dtype] - output ndarray data type
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} second argument must be an ndarray-like object
+* @throws {TypeError} options argument must be an object
+* @throws {RangeError} dimension indices must not exceed input ndarray bounds
+* @throws {RangeError} number of dimension indices must not exceed the number of input ndarray dimensions
+* @throws {Error} must provide valid options
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var Uint8Array = require( '@stdlib/array/uint8' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+*
+* // Create a mask buffer:
+* var mbuf = new Uint8Array( [ 0, 0, 0, 1, 0, 1 ] );
+*
+* // Define the shape of the input array:
+* var sh = [ 3, 2 ];
+*
+* // Define the array strides:
+* var sx = [ 2, 1 ];
+* var sm = [ 2, 1 ];
+*
+* // Define the index offset:
+* var ox = 0;
+* var om = 0;
+*
+* // Create the input ndarray:
+* var x = new ndarray( 'float64', xbuf, sh, sx, ox, 'row-major' );
+*
+* // Create the mask ndarray:
+* var mask = new ndarray( 'uint8', mbuf, sh, sm, om, 'row-major' ); // cspell:disable-line
+*
+* // Perform reduction:
+* var out = mskmax( x, mask );
+* // returns <ndarray>
+*
+* var v = out.get();
+* // returns 5.0
+*/
+function mskmax( x, mask ) {
+	var arrs;
+	if ( !isNdarrayLike( x ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be an ndarray-like object. Value: `%s`.', x ) );
+	}
+	if ( !isNdarrayLike( mask ) ) {
+		throw new TypeError( format( 'invalid argument. Second argument must be an ndarray-like object. Value: `%s`.', mask ) );
+	}
+	arrs = maybeBroadcastArrays( [ x, mask ] );
```

Comment:
Handle the options argument if provided. If no `dtype` option specified, explicitly set the `dtype` option to the dtype of `x`. This will ensure that the return type is always the same as the input ndarray, by default.